File: sample/src/main/java/net/mm2d/upnp/sample/StateVariableNode.java
/*
 * Copyright (c) 2016 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.sample;

import net.mm2d.upnp.StateVariable;

/**
 * @author <a href="mailto:ryo@mm2d.net">大前良介 (OHMAE Ryosuke)</a>
 */
public class StateVariableNode extends UpnpNode {
    public StateVariableNode(final StateVariable variable) {
        super(variable);
        setAllowsChildren(false);
    }

    @Override
    public StateVariable getUserObject() {
        return (StateVariable) super.getUserObject();
    }

    @Override
    public String formatDescription() {
        return Formatter.format(getUserObject());
    }

    @Override
    public String toString() {
        return getUserObject().getName();
    }
}

File: sample/src/main/java/net/mm2d/upnp/sample/StateVariableNode.kt
/*
 * Copyright (c) 2016 大前良介 (OHMAE Ryosuke)
 *
 * This software is released under the MIT License.
 * http://opensource.org/licenses/MIT
 */

package net.mm2d.upnp.sample

import net.mm2d.upnp.StateVariable

class StateVariableNode(variable: StateVariable) : UpnpNode(variable) {
    init {
        setAllowsChildren(false)
    }

    override fun getUserObject(): StateVariable = super.getUserObject() as StateVariable
    override fun formatDescription(): String = Formatter.format(getUserObject())
    override fun toString(): String = getUserObject().name
}
